### PR TITLE
Add basic unit tests

### PR DESCRIPTION
**Pull request by jeanpimentel**

### CODE REVIEW

File: app/src/test/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    @Test
+    fun loadModels() {
+        presenter.loadModels()
+
+        verify(view).showLoading()
+        verify(interactor).loadModels(any(), any())
+    }
+
+    @Test
+    fun loadModelsTreatingError() {
```
Comment (ivamluz) [inline app/src/test/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionPresenter.kt]: I think "Handling" sounds better than "Treating"

File: app/src/test/java/com/ciandt/dragonfly/example/features/realtime/RealTimePresenterTest.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        ).thenReturn(false)
+
+        presenter.attachView(view)
+
+        verify(view).checkRealTimeRequiredPermissions()
+
+
+        presenter.onRealTimePermissionsDenied(false)
+
+        verify(preferencesRepository, never()).putBoolean(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED, true)
```
Comment (ivamluz) [inline app/src/test/java/com/ciandt/dragonfly/example/features/realtime/RealTimePresenterTest.kt:78]: Shouldn't the value be set as false?

**Review by ivamluz — APPROVED** [review]
Approved with 1 question and 1 suggestion.

File: app/src/test/java/com/ciandt/dragonfly/example/features/realtime/RealTimePresenterTest.kt (hunk 2 above)
Comment (jeanpimentel) [reply app/src/test/java/com/ciandt/dragonfly/example/features/realtime/RealTimePresenterTest.kt:78]: Line 76: Permissions were requested and the user denied, but not permanently. `onRealTimePermissionsDenied(permanently: Boolean)`

Now, at line 78, we check that `preferencesRepository.putBoolean(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED, true)` is **never** called.

File: app/src/test/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionPresenter.kt (hunk 1 above)
Comment (jeanpimentel) [reply app/src/test/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionPresenter.kt]: Done

**Pull request merged**